internal/exec/engine.go

```diff
@@ -274,8 +276,21 @@ func (e *Engine) fetchReferencedConfig(cfgRef types.ConfigReference) (types.Conf
 	if err != nil {
 		return types.Config{}, err
 	}
+	headers := http.Header{}
+	for headerName, headerValue := range resource.ConfigHeaders {
+		headers.Add(headerName, headerValue[0])
+	}
```

Comment:
Try [`headers := resource.ConfigHeaders.Clone()`](https://golang.org/pkg/net/http/#Header.Clone) :)

Reply:
Introduced in golang 1.13, CI still uses 1.11 for building

Reply:
In that case, in order to avoid truncating duplicate keys, I suggest to accommodate the whole original slice of values in the new `headers`, by manipulating the map instead of using the single-value `Add` method:
```
headers[headerName] = headerValue
```

Reply:
yes, makes sense :) thanks for the advice!
done

Reply:
@bgilbert no, `headerValue` is a string slice with all the header values, so we do the right thing here